=== modified file 'openshot/classes/sequences.py'
--- openshot/classes/sequences.py	2012-09-07 20:37:38 +0000
+++ openshot/classes/sequences.py	2013-08-20 20:42:30 +0000
@@ -152,6 +152,12 @@
 
 	def Render(self):
 
+		# Disable window redraws temporarily (prevents flicker)
+		self.project.form.MyCanvas_Left.window.freeze_updates()
+		self.project.form.TimelineCanvas_Left.window.freeze_updates()
+		self.project.form.MyCanvas.window.freeze_updates()
+		self.project.form.TimelineCanvas_Right.window.freeze_updates()
+
 		# Clear the canvases
 		self.project.form.MyCanvas_Left.set_root_item(goocanvas.Group())
 		self.project.form.TimelineCanvas_Left.set_root_item(goocanvas.Group())
@@ -180,7 +186,11 @@
 				# Render track			
 				MyTran.Render()
 
-
+		# Enable window redraws again
+		self.project.form.MyCanvas_Left.window.thaw_updates()
+		self.project.form.TimelineCanvas_Left.window.thaw_updates()
+		self.project.form.MyCanvas.window.thaw_updates()
+		self.project.form.TimelineCanvas_Right.window.thaw_updates()
 
 	def GenerateXML(self, dom, xmlParentNode):
 

=== modified file 'openshot/windows/ui/Main.ui'
--- openshot/windows/ui/Main.ui	2012-04-29 21:18:25 +0000
+++ openshot/windows/ui/Main.ui	2013-08-20 20:42:30 +0000
@@ -1514,7 +1514,7 @@
                                 <property name="visible">True</property>
                                 <property name="can_focus">True</property>
                                 <property name="tooltip_text" translatable="yes">Slide to Zoom</property>
-                                <property name="update_policy">delayed</property>
+                                <property name="update_policy">continuous</property>
                                 <property name="adjustment">adjustment3</property>
                                 <property name="draw_value">False</property>
                                 <signal name="change-value" handler="on_hsZoom_change_value" swapped="no"/>

