--- openshot-1.4.3/openshot/classes/video.py~	2012-09-30 22:33:29.000000000 +0200
+++ openshot-1.4.3/openshot/classes/video.py	2014-09-01 08:43:10.601932534 +0200
@@ -154,6 +154,7 @@
 					self.c.set("subcmp", "2")
 					self.c.set("cmp", "2")
 					self.c.set("coder", "1")
+					self.c.set("crf", "18")
 					self.c.set("flags", "+loop")
 					self.c.set("flags2", "dct8x8")
 					self.c.set("qmax", "51")
